### PR TITLE
builtin: add oct built-in function

### PR DESCRIPTION
**Pull request by silenceper**

### CODE REVIEW

File: plan/typeinferer.go [hunk 1]
```diff
@@ -404,6 +404,8 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 		tp = types.NewFieldType(mysql.TypeVarString)
 		chs = v.defaultCharset
 		tp.Flen = 40
+	case ast.Oct:
+		tp = types.NewFieldType(mysql.TypeVarString)
```
Comment (coocood) [inline plan/typeinferer.go]: Please set charset.

File: expression/builtin_string_test.go [hunk 2]
```diff
@@ -949,3 +949,36 @@ func (s *testEvaluatorSuite) TestField(c *C) {
...
+		ret    string
+	}{
+		{"0", "0"},
+		{"1", "1"},
+		{"8", "10"},
+		{"12", "14"},
+		{"20", "24"},
+		{"100", "144"},
+		{"1024", "2000"},
+		{"2048", "4000"},
```
Comment (coocood) [inline expression/builtin_string_test.go]: Please add cases for different origin types like integer and float.

Comment (silenceper) [inline expression/builtin_string_test.go]: Right, add some negative numbers

**Comment by silenceper** [comment]
fixed and please review again.
@coocood

File: expression/builtin_string_test.go (hunk 2 above)
Comment (coocood) [inline expression/builtin_string_test.go]: Please add a string that is larger than max uint64.

File: expression/builtin_string.go [hunk 3]
```diff
@@ -1412,7 +1413,35 @@ type builtinOctSig struct {
...
+	n = getValidPrefix(strings.TrimSpace(n), 10)
+	if len(n) == 0 {
+		d.SetString("0")
+		return d, nil
+	}
+	if n[0] == '-' {
+		negative = true
+		n = n[1:]
+	}
+	val, err := strconv.ParseUint(n, 10, 64)
```
Comment (coocood) [inline expression/builtin_string.go]: The error is discarded.

File: expression/builtin_string_test.go (hunk 2 above)
Comment (silenceper) [inline expression/builtin_string_test.go]: fixed.
should output the maximum value after exceeding unint64.

**Comment by coocood** [comment]
LGTM
Thank you for your PR!

**Comment by XuHuaiyu** [comment]
@silenceper 
If possible, please check and register the builtin function you decide to implement in [issue](https://github.com/pingcap/tidb/issues/2863) to let others know, thank you!

**Comment by zimulala** [comment]
LGTM

**Pull request merged**